### PR TITLE
harden FileUtil against aliases

### LINKED ISSUE #304: Move FileUtil to com.google.cloud.tools.io
java/com/google/cloud/tools/appengine/cloudsdk/internal/FileUtil.java

### PR DESCRIPTION
**Pull request by elharo**
@meltsufin @loosebazooka fix #304

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/io/FileUtil.java [hunk 1]
```diff
@@ -28,23 +29,28 @@
...
   public static void copyDirectory(final Path source, final Path destination) throws IOException {
     Preconditions.checkNotNull(source);
     Preconditions.checkNotNull(destination);
     Preconditions.checkArgument(Files.isDirectory(source));
     Preconditions.checkArgument(Files.isDirectory(destination));
-    Preconditions.checkArgument(!source.equals(destination));
-    Preconditions.checkArgument(!destination.startsWith(source), "destination is child of source");
+    Preconditions.checkArgument(!Files.isSameFile(source, destination),
+        "Source and destination are the same");
+    Preconditions.checkArgument(!destination.toAbsolutePath().startsWith(source.toAbsolutePath()),
```
Comment (patflynn) [inline src/main/java/com/google/cloud/tools/io/FileUtil.java]: Would this copy fail?
source: /path/funny (is a directory)
destination: /path/funny2

File: src/main/java/com/google/cloud/tools/io/FileUtil.java [hunk 2]
```diff
@@ -28,23 +29,28 @@
 import java.nio.file.attribute.BasicFileAttributes;
 
 /**
- * Internal file utilities.
+ * File utilities we hope to eventually push up to Guava.
```
Comment (patflynn) [inline src/main/java/com/google/cloud/tools/io/FileUtil.java]: I'd love to talk about pushing this to Guava, but I'm not convinced it's valuable as a public javadoc on the class.

File: src/main/java/com/google/cloud/tools/io/FileUtil.java (hunk 1 above)
Comment (briandealwis) [inline src/main/java/com/google/cloud/tools/io/FileUtil.java]: `Path.startsWith()` describes as performing an element-by-element comparison.

Comment (elharo) [inline src/main/java/com/google/cloud/tools/io/FileUtil.java]: No, tests added to prove it

File: src/main/java/com/google/cloud/tools/io/FileUtil.java (hunk 2 above)
Comment (elharo) [inline src/main/java/com/google/cloud/tools/io/FileUtil.java]: done

**Comment by elharo** [comment]
Ping

**Pull request merged**